### PR TITLE
fix: set the default ttl to 60s

### PR DESCRIPTION
**Pull request by zhiying-lin**
**What type of PR is this?**

/kind bug
/kind test

**What this PR does / why we need it**:
- fix the TTL bug, atm's ttl default value is 0
- add azure client in e2e to validate azure resources

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] run `make reviewable` for basic local test
- [ ] Read and followed fleet-networking's [Code of conduct](https://github.com/Azure/fleet-networking/blob/main/CODE_OF_CONDUCT.md).
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

### How has this code been tested

e2e tests passed, https://github.com/Azure/fleet-networking/actions/runs/12463049682

### CODE REVIEW

File: test/e2e/traffic_manager_test.go [hunk 1]
```diff
@@ -43,7 +47,38 @@ var _ = Describe("Test exporting service via Azure traffic manager", func() {
 		Expect(hubClient.Create(ctx, &profile)).Should(Succeed(), "Failed to creat the trafficManagerProfile")
 
 		By("Validating the trafficManagerProfile status")
-		validator.ValidateIfTrafficManagerProfileIsProgrammed(ctx, hubClient, types.NamespacedName{Namespace: profile.Namespace, Name: profile.Name})
+		wantDNSName := validator.ValidateIfTrafficManagerProfileIsProgrammed(ctx, hubClient, types.NamespacedName{Namespace: profile.Namespace, Name: profile.Name})
+
+		By("Validating the atm profile")
```
Comment (ryanzhang-oss) [inline test/e2e/traffic_manager_test.go:52]:
```suggestion
		By("Validating the atm profile Azure resource")
```

**Review by ryanzhang-oss — APPROVED** [review]

File: pkg/controllers/hub/trafficmanagerprofile/controller.go [hunk 2]
```diff
@@ -271,6 +280,7 @@ func generateAzureTrafficManagerProfile(profile *fleetnetv1alpha1.TrafficManager
 		Properties: &armtrafficmanager.ProfileProperties{
 			DNSConfig: &armtrafficmanager.DNSConfig{
 				RelativeName: ptr.To(fmt.Sprintf(DNSRelativeNameFormat, profile.Namespace, profile.Name)),
+				TTL:          ptr.To(DefaultDNSTTL), // no default value on the server side, using 60s same as portal's default config
```
Comment (jwtty) [inline pkg/controllers/hub/trafficmanagerprofile/controller.go:283]: This is not configurable from current API right?

File: pkg/controllers/hub/trafficmanagerprofile/controller.go [hunk 3]
```diff
@@ -170,8 +175,8 @@ func (r *Reconciler) handleUpdate(ctx context.Context, profile *fleetnetv1alpha1
 // by ignoring others.
 // The desired profile is built by the controllers and all the required fields should not be nil.
 func EqualAzureTrafficManagerProfile(current, desired armtrafficmanager.Profile) bool {
-	// location and dnsConfig is not immutable
-	if current.Properties == nil || current.Properties.MonitorConfig == nil || current.Properties.ProfileStatus == nil || current.Properties.TrafficRoutingMethod == nil {
+	// location and dnsConfig (excluding TTL) is not immutable
```
Comment (jwtty) [inline pkg/controllers/hub/trafficmanagerprofile/controller.go:178]: I don't understand this comment.

File: test/e2e/e2e_test.go [hunk 4]
```diff
@@ -67,8 +78,29 @@ var _ = BeforeSuite(func() {
...
 })
 
+func initAzureClients() {
+	subscriptionID := os.Getenv(azureSubscriptionEnv)
+	Expect(subscriptionID).ShouldNot(BeEmpty(), "Azure subscription ID is not set")
+
+	atmResourceGroup := os.Getenv(azureTrafficManagerResourceGroupEnv)
+	Expect(atmResourceGroup).ShouldNot(BeEmpty(), "Azure traffic manager resource group is not set")
+
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```
Comment (jwtty) [inline test/e2e/e2e_test.go:92]: Is this the way to use the federated identity? Do we use the same identity that provisions the aks clusters for existing e2e tests?

File: test/e2e/framework/workload_manager.go [hunk 5]
```diff
@@ -138,7 +138,12 @@ func (wm *WorkloadManager) TrafficManagerProfile() fleetnetv1alpha1.TrafficManag
 		},
 		Spec: fleetnetv1alpha1.TrafficManagerProfileSpec{
 			MonitorConfig: &fleetnetv1alpha1.MonitorConfig{
-				Port: ptr.To(int64(80)),
+				Port:                      ptr.To(int64(80)),
```
Comment (jwtty) [inline test/e2e/framework/workload_manager.go:141]: Why do we need to set the additional fields?

File: test/common/trafficmanager/azureprovider/profile.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	cmpProfileOptions = cmp.Options{
+		cmpopts.IgnoreFields(armtrafficmanager.Profile{}, "ID", "Name", "Type"),
+		cmpopts.IgnoreFields(armtrafficmanager.MonitorConfig{}, "ProfileMonitorStatus"), // cannot predict the monitor status
+	}
+)
+
+// Validator contains the way of accessing the Azure Traffic Manager resources.
+type Validator struct {
+	ProfileClient  *armtrafficmanager.ProfilesClient
+	EndpointClient *armtrafficmanager.EndpointsClient
```
[the remark this answers is not in the document]
Comment (zhiying-lin) [reply test/common/trafficmanager/azureprovider/profile.go]: removed, will add back when needed

File: pkg/controllers/hub/trafficmanagerprofile/controller.go (hunk 2 above)
Comment (zhiying-lin) [reply pkg/controllers/hub/trafficmanagerprofile/controller.go:284]: Yes, we don't allow cx to config for now.

File: pkg/controllers/hub/trafficmanagerprofile/controller.go (hunk 3 above)
Comment (zhiying-lin) [reply pkg/controllers/hub/trafficmanagerprofile/controller.go]: typo

File: test/e2e/e2e_test.go (hunk 4 above)
Comment (zhiying-lin) [reply test/e2e/e2e_test.go:92]: the login is part of the github action, https://github.com/Azure/fleet-networking/blob/main/.github/workflows/e2e-tests.yml#L66

Yes, we use the same identity for the setup scripts.

This func will automatically use the identity which you've already logged in.

File: test/e2e/framework/workload_manager.go (hunk 5 above)
Comment (zhiying-lin) [reply test/e2e/framework/workload_manager.go:141]: to make sure our controller will set these fields

**Pull request merged**